### PR TITLE
Fix usage of `opts` in Api and SSO module

### PR DESCRIPTION
**Pull request by jacobiajohnson**

### CODE REVIEW

**Comment by jacobiajohnson** [comment]
Current dependencies on/for this PR:
* main
  * **PR #21** <a href="https://app.graphite.dev/github/pr/workos/workos-elixir/21" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/workos/workos-elixir/21?utm_source=stack-comment).

**Review by blairlunceford — APPROVED** [review]

File: lib/workos/api.ex [hunk 1]
```diff
@@ -8,9 +8,11 @@ defmodule WorkOS.API do
   """
   def client(opts \\ []) do
     auth = WorkOS.api_key(opts)
-    if is_map_key(opts, :access_token) do
-      auth = opts[:access_token]
+
+    if opts[:access_token] do
+      ^auth = opts[:access_token]
```
Comment (seantanly) [inline lib/workos/api.ex:13]: Is there a wrong assertion here, that access_token should be the same as API key?
This seems to be breaking the `SSO.get_profile` function.

@jacobiajohnson @blairlunceford

File: lib/workos/sso/sso.ex [hunk 2]
```diff
@@ -94,7 +94,7 @@ defmodule WorkOS.SSO do
     access_token = %{
```
Comment (seantanly) [inline lib/workos/sso/sso.ex:94]: `access_token` above is a map, should not be concatenated to a keyword list. It is producing a wrong result
```
iex> [foo: "bar"] ++ %{access_token: 123}
[{:foo, "bar"} | %{access_token: 123}]
```

The section can be rewritten as
```
opts = opts ++ [access_token: access_token]
```
or simply
```
opts = opts |> Keyword.put(:access_token, access_token)
```

@jacobiajohnson @blairlunceford

**Review by seantanly — COMMENTED** [review]
@jacobiajohnson @blairlunceford

This PR has introduced a regression in the SSO.get_profile function, and the fix on opts usage isn't exactly right.

Your kind attention please, thank you.

**Pull request merged**